File: nishant_mishra/day_3/two_sum.cpp
// https://leetcode.com/problems/two-sum/

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map <int,int> m;
        vector<int> result(2,0);
        for (int i = 0; i < nums.size(); ++i) {
            int find = target - nums[i];
            if (m.find(find) != m.end()) {
                result[0] = i;
                result[1] = m[find];
                return result;
            }
            m[nums[i]] = i;
        }
        return result;
    }
};

File: Akshat Jain/Day 8/two_sum.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        l,r=0,len(numbers)-1
        while l<r:
            cursum=numbers[l]+numbers[r]
            if cursum>target:
                r-=1
            elif cursum<target:
                l+=1
            elif cursum==target:
                return [l+1,r+1]